packages/@aws-cdk/aws-location-alpha/lib/geofence-collection.ts

```diff
@@ -0,0 +1,165 @@
+
+  /**
+   * Grant the given principal identity permissions to perform the actions on this geofence collection.
+   */
+  public grant(grantee: iam.IGrantable, ...actions: string[]): iam.Grant {
+    return iam.Grant.addToPrincipal({
+      grantee: grantee,
+      actions: actions,
+      resourceArns: [this.geofenceCollectionArn],
+    });
+  }
+
+  /**
+   * Grant the given identity permissions to read this geofence collection
+   *
+   * @see https://docs.aws.amazon.com/location/latest/developerguide/security_iam_id-based-policy-examples.html#security_iam_id-based-policy-examples-read-only-geofences
+   */
+  public grantRead(grantee: iam.IGrantable): iam.Grant {
+    return this.grant(grantee,
+      'geo:ListGeofences',
+      'geo:GetGeofence',
+    );
+  }
+}
+
+/**
+ * A Geofence Collection
+ *
+ * @see https://docs.aws.amazon.com/location/latest/developerguide/geofence-tracker-concepts.html#geofence-overview
+ */
+export class GeofenceCollection extends GeofenceCollectionBase {
+  /**
+   * Use an existing geofence collection by name
+   */
+  public static fromGeofenceCollectionName(scope: Construct, id: string, geofenceCollectionName: string): IGeofenceCollection {
+    const geofenceCollectionArn = Stack.of(scope).formatArn({
+      service: 'geo',
+      resource: 'geofence-collection',
+      resourceName: geofenceCollectionName,
+    });
+
+    return GeofenceCollection.fromGeofenceCollectionArn(scope, id, geofenceCollectionArn);
+  }
+
+  /**
+   * Use an existing geofence collection by ARN
+   */
+  public static fromGeofenceCollectionArn(scope: Construct, id: string, geofenceCollectionArn: string): IGeofenceCollection {
+    const parsedArn = Stack.of(scope).splitArn(geofenceCollectionArn, ArnFormat.SLASH_RESOURCE_NAME);
+
+    if (!parsedArn.resourceName) {
+      throw new Error(`Geofence Collection Arn ${geofenceCollectionArn} does not have a resource name.`);
+    }
+
+    class Import extends GeofenceCollectionBase {
+      public readonly geofenceCollectionName = parsedArn.resourceName!;
+      public readonly geofenceCollectionArn = geofenceCollectionArn;
+    }
+
+    return new Import(scope, id, {
+      account: parsedArn.account,
+      region: parsedArn.region,
+    });
+  }
+
+  public readonly geofenceCollectionName: string;
+
+  public readonly geofenceCollectionArn: string;
+
+  /**
+   * The timestamp for when the geofence collection resource was created in ISO 8601 format
+   *
+   * @attribute
+   */
+  public readonly geofenceCollectionCreateTime: string;
+
+  /**
+   * The timestamp for when the geofence collection resource was last updated in ISO 8601 format
+   *
+   * @attribute
+   */
+  public readonly geofenceCollectionUpdateTime: string;
+
+  constructor(scope: Construct, id: string, props: GeofenceCollectionProps = {}) {
+
+    if (props.description !== undefined && !Token.isUnresolved(props.description) && props.description.length > 1000) {
```

Comment:
```suggestion
    if (props.description && !Token.isUnresolved(props.description) && props.description.length > 1000) {
```
Not needed given that `""` is legit.